nexus/db-queries/src/db/datastore/deployment.rs

```diff
@@ -1632,10 +1595,23 @@ mod tests {
         let blueprint2 = builder.build();
         let authz_blueprint2 = authz_blueprint_from_id(blueprint2.id);
 
+        let diff = blueprint2.diff_since_blueprint(&blueprint1).unwrap();
+        println!("b1 -> b2: {}", diff.display());
+        println!("b1 disks: {:?}", blueprint1.blueprint_disks);
+        println!("b2 disks: {:?}", blueprint2.blueprint_disks);
         // Check that we added the new sled, as well as its disks and zones.
         assert_eq!(
-            blueprint1.blueprint_disks.len() + new_sled_zpools.len(),
-            blueprint2.blueprint_disks.len(),
+            blueprint1
+                .blueprint_disks
+                .values()
+                .map(|c| c.disks.len())
+                .sum::<usize>()
+                + new_sled_zpools.len(),
+            blueprint2
+                .blueprint_disks
+                .values()
+                .map(|c| c.disks.len())
+                .sum::<usize>()
```

Comment:
This assertion change is, I believe, a fix for a test that was passing incorrectly. `blueprint_disks` is a `BTreeMap<SledUuid, PhysicalDisksConfig>`, which means `blueprint_disks.len()` reported "for how many sleds do we have a `PhysicalDisksConfig`?", not "how many disks are in the blueprint" as the check seemed to mean. I think this was passing purely by coincidence:

* `blueprint1.blueprint_disks` was empty, because a blueprint created from an inventory collection didn't populate it at all
* `new_sled_zpools` was 4, because we added 4 disks
* `blueprint2.blueprint_disks.len()` was 4, because there are 4 sleds in blueprint2, and going through the builder populates `blueprint_disks`

so we were previously passing because `0 + 4 == 4`, even though the units of these values made the comparison nonsensical.